spec/models/grace_period_submission_rule_spec.rb

```diff
@@ -40,6 +40,60 @@
       destroy_repos
     end
 
+    context 'when the student did not submit any files' do
+      # Before the due_date and before collection_time
+      describe 'when assignment is collected before the due date' do
+        before :each do
+          @submission = create(:submission, grouping: @grouping, is_empty: true)
```

Comment:
After looking at this a bit more, I'm worried that just by creating the submission here, `apply_submission_rule` isn't actually being applied. Look at how it's done in the existing tests on lines 119-120; that's probably what you should do here.

Reply:
The rule is being applied for the corresponding submission at no_submission_file.rb.

Reply:
Okay that's fine (and it's good you added it to the other shared example).